files/default/bin/deploy.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+PUBLIC=0
+
+run_help() {
+  echo
+  tput bold
+  echo " Name:"
+  tput sgr0
+  echo "    $0 - deploy target image to OpenStack for further testing."
+  echo
+  tput bold
+  echo " Usage:"
+  tput sgr0
+  echo "    $0 [-p] -f [FILE] -n [IMG_NAME] -r [PR_NUM]"
+  echo
+  tput bold
+  echo " Options:"
+  tput sgr0
+  echo "    -p          - Publish (publicly) the target image as '\$IMG_NAME', renaming the old"
+  echo "                  '\$IMG_NAME' to '\$IMG_NAME - \$PR_NUM' and making that image private"
+  echo
+  tput bold
+  echo " Arguments:"
+  tput sgr0
+  echo
+  echo "    FILE        - Fully built image to be uploaded to OpenStack if [-p] is"
+  echo "                  not specified. The image will be: '\$IMG_NAME - \$PR_NUM'"
+  echo
+  echo "    IMG_NAME    - Name of the image to replace if [-p] is specified."
+  echo
+  echo "    PR_NUM      - GitHub Pull Request number to name private images."
+  echo
+  exit 0
+}
+
+while getopts "hpf:n:r:" opt ; do
+  case $opt in
+    h)
+      run_help
+      ;;
+    p)
+      PUBLIC=1
+      ;;
+    f)
+      FILE="$OPTARG"
+      ;;
+    n)
+      IMG_NAME="$OPTARG"
+      ;;
+    r)
+      PR_NUM="$OPTARG"
+      ;;
+  esac
+done
+
+set -xe
```

Comment:
Is this a debug leftover ? If not, you might want to move this after the error messages.